File: backend/src/main/java/group8/bloodbank/model/AppointmentHistory.java
package group8.bloodbank.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDateTime;


@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class AppointmentHistory {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @ManyToOne
    @JoinColumn(name = "donor_id")
    private Donor donor;
    @ManyToOne
    @JoinColumn(name = "blood_bank_id")
    private BloodBank bloodBank;
    @ManyToOne
    @JoinColumn(name = "medical_worker_id")
    private MedicalWorker medicalWorker;
    @Column
    private LocalDateTime date;
    @Column
    private BloodType bloodType;
    @Column
    private double amount;
    @Column
    private AppointmentStatus status;
}

File: frontend/src/app/model/appointmentHistory.ts
import { AppointmentStatus } from "./AppointmentStatus";
import { BloodType } from "./bloodType";
import { Item } from "./item";

export class AppointmentHistory{
    id!: number;
    bloodBankId!: number;
    date!: Date;
    donorId!: number;
    medicalWorkerId!: number;
    bloodType!: string;
    amount!: number;
    status!: AppointmentStatus
    item!: Item[]
    appointmentId!: number;

    constructor(obj?: any){
        if(obj) {
            this.id = obj.id;
            this.bloodBankId = obj.bloodBankId;
            this.date = obj.date;
            this.donorId = obj.donorId;
            this.medicalWorkerId = obj.medicalWorkerId;
            this.bloodType = obj.bloodType;
            this.amount = obj.amount;
            this.status = obj.status;
            this.item = obj.item;
            this.appointmentId = obj.appointmentId;
        }
    }
}